File: Code/5.py
# A. Beautiful Matrix - https://codeforces.com/contest/263/problem/A

def solve(arr):
    pos = arr.index(1)+1
    steps = 0
    if pos == 13:
        return 0
    while (pos != 13):
        if (pos >= 1 and pos <= 10):
            pos += 5
            steps += 1
        elif (pos >= 11 and pos < 13):
            pos += 1
            steps += 1
        elif (pos > 13 and pos <= 15):
            pos -= 1
            steps += 1
        elif (pos >= 16 and pos <= 25):
            pos -= 5
            steps += 1
    return steps

# arr = [list(map(int, input().split())) for _ in range(5)]
grid = [list(map(int, input().split())) for _ in range(5)]
arr = [element for row in grid for element in row]


print(solve(arr))

File: Code/5.cpp
/* 29-Oct-2023 -  A. Beautiful Matrix -
 * https://codeforces.com/contest/263/problem/A */

#include <iostream>
using namespace std;

int main() {
  int x = 0;
  auto k = ((5 + 1) / 2); // Middle of 5x5 Matrix. n+1/2 of nxn matrix

  // find pos of 1
  for (int i = 1; i <= 5; i++) {
    for (int j = 1; j <= 5; j++) {
      cin >> x;
      if (x == 1) {
        cout << abs(i - k) + abs(j - k);
        break;
      }
    }
  }
  return 0;
}

/* In a 7x7 grid, k = 7 + 1/2 = 4, so the center is at (4,4) */
/* center is at (k,k) */
/* Manhattan Distance=∣x−k∣+∣y−k∣ */
